arches/app/models/graph.py

```diff
@@ -1377,14 +1377,17 @@ def find_unpermitted_edits(obj_a, obj_b, ignore_list, obj_type):
                 unpermitted_edits = []
                 db_nodes = models.Node.objects.filter(graph=self)
                 for db_node in db_nodes:
-                    unpermitted_node_edits = find_unpermitted_edits(
-                        db_node,
-                        self.nodes[db_node.nodeid],
-                        ["name", "issearchable", "ontologyclass", "description", "isrequired", "fieldname", "exportable"],
-                        "node",
-                    )
-                    if unpermitted_node_edits is not None:
-                        unpermitted_edits.append(unpermitted_node_edits)
+                    try:
+                        unpermitted_node_edits = find_unpermitted_edits(
+                            db_node,
+                            self.nodes[db_node.nodeid],
+                            ["name", "issearchable", "ontologyclass", "description", "isrequired", "fieldname", "exportable"],
+                            "node",
+                        )
+                        if unpermitted_node_edits is not None:
+                            unpermitted_edits.append(unpermitted_node_edits)
+                    except KeyError:
```

Comment:
what causes this to error?

Reply:
This was actually the result of a corrupt graph I had - where there was no db_node.nodeid in self.nodes.  I'll remove this; as it should never happen.